### PR TITLE
Refactory `ClientInterface` & `Client` class

### LINKED ISSUE #20: Update all API clent methods parameters (a lot of params -> single params object)
Right now all API client methods (except `::userReportMake`) requires a lot of different parameners to invoke. The better strategy is using **single** parameters object (like in #17).

We need to update all methods before `v4.x` releasing.

### PR DESCRIPTION
**Pull request by gomzyakov**
## Changed

- Refactory `ClientInterface` & `Client` class

Fixes #20 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -173,30 +184,23 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
-                    '_query'        => $query,
-                    '_size'         => \max(1, $size),
-                    '_offset'       => \max(0, $offset),
-                    '_page'         => \max(1, $page),
-                    '_sort'         => $sort,
-                    '_calc_total'   => $calc_total
+                    '_query'        => $params->getQuery(),
+                    '_size'         => \max(1, $params->getPerPage()),
+                    '_offset'       => \max(0, $params->getOffset()),
+                    '_page'         => \max(1, $params->getPage()),
```
Comment (gomzyakov) [inline src/Client.php]: @tarampampam @eldario

Potentially, we can transmit an incorrect combination of `_size`, `_offset` & `_page`. Check combination -- Is this `b2bapi` responsibility?

File: src/Params/UserReportParams.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * @var bool
+     */
+    private $include_content = true;
+
+    /**
+     * True, if necessary detailed information about report.
+     *
+     * @var bool
+     */
+    private $detailed = true;
```
Comment (gomzyakov) [inline src/Params/UserReportParams.php]: @tarampampam @eldario

Similar pairs of getters/setters can be taken out **to traits**. But, I doubt... 🤔

File: src/ClientInterface.php [hunk 3]
```diff
@@ -21,151 +28,108 @@ interface ClientInterface
...
-     * @param string|null $value Any string value (server must returns it back)
+     * @param DevPingParams $params
      *
      * @throws BadRequestException
      * @throws BadResponseException
      *
      * @return DevPingResponse
      */
-    public function devPing(?string $value = null): DevPingResponse;
+    public function devPing(DevPingParams $params): DevPingResponse;
```
Comment (eldario) [inline src/ClientInterface.php]: Specifically in such cases, is it possible to make the parameter `DevPingParams $params` nullable?

```php
 public function devPing(?DevPingParams $params): DevPingResponse;
```

For simple use like in older version.

```php
/** #var ClientInterface $client */
$client->devPing();
```

File: src/ClientInterface.php [hunk 4]
```diff
@@ -21,151 +28,108 @@ interface ClientInterface
...
-     * @param bool $detailed
+     * @param UserParams $params
      *
      * @throws BadRequestException
      * @throws BadResponseException
      *
      * @return UserResponse
      */
-    public function user(bool $detailed = false): UserResponse;
+    public function user(UserParams $params): UserResponse;
```
Comment (eldario) [inline src/ClientInterface.php]: Also in this method 
```php
user(?UserParams $params)
```

File: src/Params/AbstractListParams.php [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+     * @var string
+     */
+    private $sort_by = '-created_at';
+
+    /**
+     * True, if necessary include list item content into response.
+     *
+     * @var bool
+     */
+    private $include_content = false;
```
Comment (eldario) [inline src/Params/AbstractListParams.php]: In my opinion `with_content` is better than `include_content`

File: src/Params/DevPingParams.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Avtocod\B2BApi\Params;
+
+final class DevPingParams
```
Comment (eldario) [inline src/Params/DevPingParams.php]: Maybe transfer the prefix `Dev` to the namespace?

Like:
```php
namespace Avtocod\B2BApi\Params\Dev;

final class PingParams{
...
```

Or

```php
namespace Avtocod\B2BApi\Params\User;

final class BalanceParams{
...

File: src/Params/DevTokenParams.php [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    {
+        return $this->date_from;
+    }
+
+    /**
+     * @param DateTimeImmutable|null $date_from
+     *
+     * @return $this
+     */
+    public function setDateFrom(?DateTimeImmutable $date_from): self
```
Comment (eldario) [inline src/Params/DevTokenParams.php]: Why nullable?

File: src/Params/UserReportRefreshParams.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+     * Report unique ID (e.g.: `some_report_uid_YV1KS9614S107357Y@domain`).
+     *
+     * @var string
+     */
+    private $report_uid;
+
+    /**
+     * Additional request options.
+     *
+     * @var array<string, string|int|float|bool|array<mixed>>
```
Comment (eldario) [inline src/Params/UserReportRefreshParams.php]: Forget about `|null`

**Review by eldario — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,14 +6,13 @@ The format is based on [Keep a Changelog][keepachangelog] and this project adher
...
 
-- Class `ReportMakeParams` to build make-report parameters
-- Optional parameter `idempotenceKey` for report-make requests
+- Refactory `ClientInterface` & `Client` class
 
-### Changed
+### Added
 
-- Method `ClientInterface::userReportMake` now takes object `ReportMakeParams` as parameter
+- Optional parameter `idempotenceKey` in `ReportMakeParams` for report-make requests
```
Comment (tarampampam) [inline CHANGELOG.md]: All is ok here?

File: README.md [hunk 10]
```diff
@@ -49,47 +49,64 @@ $client = new Client(new Settings(AuthToken::generate('username', 'password', 'd
...
+use \Avtocod\B2BApi\Client;
+use \Avtocod\B2BApi\Params\DevPingParams;
+use \Avtocod\B2BApi\Params\DevTokenParams;
+use \Avtocod\B2BApi\Params\UserParams;
+use \Avtocod\B2BApi\Params\UserBalanceParams;
+use \Avtocod\B2BApi\Params\UserReportTypesParams;
+use \Avtocod\B2BApi\Params\UserReportsParams;
+use \Avtocod\B2BApi\Params\UserReportParams;
+use \Avtocod\B2BApi\Params\UserReportMakeParams;
+use \Avtocod\B2BApi\Params\UserReportRefreshParams;
```
Comment (tarampampam) [inline README.md]: As I can see - this classes uses just once, so, `use` usage is required here? Maybe use "inline" instead?

File: README.md [hunk 11]
```diff
@@ -49,47 +49,64 @@ $client = new Client(new Settings(AuthToken::generate('username', 'password', 'd
...
+
+/** @var Client $client */
 
 // Test connection
-$client->devPing();
+$client->devPing(new DevPingParams());
```
Comment (tarampampam) [inline README.md]:
```suggestion
$client->devPing(new DevPingParams);
```

You can ommit `()` here and below

File: README.md (hunk 10 above)
Comment (tarampampam) [reply README.md]: Here and below, of course

**Review by tarampampam — CHANGES_REQUESTED** [review]

File: README.md (hunk 11 above)
Comment (gomzyakov) [reply README.md]: In all places, removed `()` if constructor is empty adbb3ff001563e430a08eca13f4bf7da54122c99

Comment (gomzyakov) [reply README.md]: Removed in all places in commit adbb3ff001563e430a08eca13f4bf7da54122c99

File: README.md (hunk 10 above)
Comment (gomzyakov) [reply README.md]: I think inline usage in examples is very heavy. Maybe not showing `use` in examples?

If we look at the Laravel [documentation](https://laravel.com/docs/8.x/eloquent), `use` sometimes used in examples, sometimes not:

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Flight extends Model
{
    /**
     * The storage format of the model's date columns.
     *
     * @var string
     */
    protected $dateFormat = 'U';
}
```

```php
<?php

class Flight extends Model
{
    const CREATED_AT = 'creation_date';
    const UPDATED_AT = 'last_update';
}
```

File: src/Params/AbstractListParams.php (hunk 5 above)
Comment (gomzyakov) [reply src/Params/AbstractListParams.php]: Yes I agree 0a090fa70454885ab612e83625b4c306d32e3652

File: src/ClientInterface.php (hunk 3 above)
Comment (gomzyakov) [reply src/ClientInterface.php]: Good idea! I have implemented this f4dbd2e360a8dd364f754e849fb875466a4dcc97

File: src/ClientInterface.php (hunk 4 above)
Comment (gomzyakov) [reply src/ClientInterface.php]: Now optional parameters are used in methods:

- `devPing()`
- `user()`
- `userReportTypes()`
- `userReports()`

File: src/Client.php [hunk 12]
```diff
@@ -117,22 +129,25 @@ public function getVersion(bool $without_hash = true): string
...
-                             int $age = 60): DevTokenResponse
+    public function devToken(DevTokenParams $params): DevTokenResponse
     {
+        $date_time_immutable = $params->getDateFrom() ?? new \DateTimeImmutable;
+
+        // Convert \DateTimeImmutable to \DateTime with timezone
+        // @notice We can use DateTime::createFromImmutable() when the PHP 7.2 support is removed
+        $date_time = new \DateTime('now', $date_time_immutable->getTimezone());
+        $date_time->setTimestamp($date_time_immutable->getTimestamp());
+
```
Comment (gomzyakov) [inline src/Client.php]: Out of scope. Need remove.

Comment (gomzyakov) [reply src/Client.php]: Rollback changes with DateTimeImmutable  932cac9bf91aa218d65c4d2f598e084367459b2c

File: src/Params/DevTokenParams.php (hunk 7 above)
Comment (gomzyakov) [reply src/Params/DevTokenParams.php]: Fixed fda3dfb394d41a2f9a19a374da7f1e9db3f59bf1

File: src/Params/UserReportRefreshParams.php (hunk 8 above)
Comment (gomzyakov) [reply src/Params/UserReportRefreshParams.php]: Thanks! Fixed b129d6e423ecb5ad39e29c5233408b43ec66c04e

File: CHANGELOG.md (hunk 9 above)
Comment (gomzyakov) [reply CHANGELOG.md:15]: Hmmm... I think yes. Am I missing something?

File: src/Params/DevPingParams.php (hunk 6 above)
Comment (gomzyakov) [reply src/Params/DevPingParams.php]: Looks great, but I doubt the name of the parameter `UserParams`.

This implementation -- is well?

```php
namespace Avtocod\B2BApi\Params\User;

final class InfoParams{
...
```

File: CHANGELOG.md (hunk 9 above)
Comment (gomzyakov) [reply CHANGELOG.md]: I think yes. Am I missing something?

**Review by eldario — APPROVED** [review]

**Comment by tarampampam** [comment]
@eldario resolve or comment all unresolved threads, please. I'll wait for it

File: src/Params/DevTokenParams.php (hunk 7 above)
Comment (eldario) [reply src/Params/DevTokenParams.php]: Ok

File: src/Params/DevPingParams.php (hunk 6 above)
Comment (eldario) [reply src/Params/DevPingParams.php]: `InfoParams` Looks pretty

File: src/Params/AbstractListParams.php (hunk 5 above)
Comment (eldario) [reply src/Params/AbstractListParams.php]: great!

File: src/ClientInterface.php (hunk 4 above)
Comment (eldario) [reply src/ClientInterface.php]: great!

File: src/ClientInterface.php (hunk 3 above)
Comment (eldario) [reply src/ClientInterface.php]: great!

File: README.md (hunk 10 above)
Comment (tarampampam) [reply README.md]: nope, use inline, please

File: CHANGELOG.md (hunk 9 above)
Comment (tarampampam) [reply CHANGELOG.md]: It seems like you forgot to describe **all** changes and broke down previous correct changelog lines

Comment (gomzyakov) [reply CHANGELOG.md]: `CHANGELOG.md` was fixed 3651963cbe79b46b6f5787b9dfea4ac096db7320

File: README.md (hunk 10 above)
Comment (gomzyakov) [reply README.md]: Okay, I'm using inline c5e64d255039475edcc38439374cf0cc03678a06

File: src/Params/UserReportParams.php (hunk 2 above)
Comment (gomzyakov) [reply src/Params/UserReportParams.php]: So far without using traits...

File: src/Client.php (hunk 1 above)
Comment (gomzyakov) [reply src/Client.php]: This is not the area of b2b-client responsibility

File: src/Params/DevPingParams.php (hunk 6 above)
Comment (gomzyakov) [reply src/Params/DevPingParams.php]: Now, params-objects with prefixes `Dev` & `User` e35274fefc31dd0d6895b14057a58f5e1d3cda62

Comment (eldario) [reply src/Params/DevPingParams.php]: Great!

**Comment by eldario** [comment]
@tarampampam @gomzyakov 
I have commented on all my **threads**

**Review by tarampampam — CHANGES_REQUESTED** [review]
Params must be thin and without any API-specific values inside. Just as an "abstract transport layer", please

File: tests/Unit/ClientTest.php [hunk 13]
```diff
@@ -879,14 +899,14 @@ public function testUserReportTypes(): void
...
             $this->settings->getBaseUri() . 'user/report_types?' . \http_build_query([
-                '_can_generate' => 'true',
                 '_content'      => 'true',
                 '_query'        => '_all',
                 '_size'         => 20,
                 '_offset'       => 0,
                 '_page'         => 1,
                 '_sort'         => '-created_at',
                 '_calc_total'   => 'true',
+                '_can_generate' => 'true',
```
Comment (gomzyakov) [inline tests/Unit/ClientTest.php]: @tarampampam 

Why is this line moved down?

This is the specifics of refactored query building. Now all common GET-parameters in the URL go first, and unique for the request at the end.

File: src/Client.php [hunk 14]
```diff
@@ -85,14 +93,18 @@ public function getSettings(): Settings
...
+            $query['value'] = $value;
+        }
+
         return DevPingResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', 'dev/ping'), [
-                'query' => [
-                    'value' => $value ?? ((string) \time()),
-                ],
-            ])
+            $this->doRequest(new Request('get', 'dev/ping'), ['query' => $query])
```
Comment (gomzyakov) [inline src/Client.php]: Now inside `doRequest(...)` we:
- Build request URL
- Store default values
- Store logic and modify options

It looks very cumbersome in my opinion.

I suggest following the scheme:
- Build query URL, if with logic
- Set default query options
- Modify options, if needed
- Send request

This is *more verbose*, but much easier to read & understand (no deep nesting).

@tarampampam do you agree or return everything inside `doRequest(...)`?

File: src/Client.php [hunk 15]
```diff
@@ -117,152 +129,180 @@ public function getVersion(bool $without_hash = true): string
...
+
+            if (is_bool($is_calc_total = $params->isCalcTotal())) {
+                $query['_calc_total'] = $is_calc_total ? 'true' : 'false';
+            }
+
+            // Unique property for `user/reports`
+            if (is_bool($can_generate = $params->isCanGenerate())) {
+                $query['_can_generate'] = $can_generate ? 'true' : 'false';
+            }
+        }
```
Comment (gomzyakov) [inline src/Client.php]: If we set some common interface (like a `ItemsListInterface`) to `UserReportsParams` and `UserReportTypesParams`, we can move two large chunks of code into a private/protected method.

Methods `userReports` and `userReportTypes` will become shorter and *more-DRY*.


@tarampampam What do you think?

**Review by tarampampam — CHANGES_REQUESTED** [review]
Do not make "query building" inside each API client method - it seems like "code for a code"

File: src/Client.php [hunk 16]
```diff
@@ -173,87 +179,113 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
                 'query' => [
-                    '_content'  => $content
-                        ? 'true'
-                        : 'false',
-                    '_detailed' => $detailed
-                        ? 'true'
-                        : 'false',
+                    '_content'  => $params->isIncludeContent() === false
+                        ? 'false'
+                        : 'true',
```
Comment (gomzyakov) [inline src/Client.php]: I have to explain why order changed? :)

File: src/Client.php [hunk 17]
```diff
@@ -85,12 +93,14 @@ public function getSettings(): Settings
...
      * {@inheritdoc}
      */
-    public function devPing(?string $value = null): DevPingResponse
+    public function devPing(?DevPingParams $params = null): DevPingResponse
     {
         return DevPingResponse::fromHttpResponse(
             $this->doRequest(new Request('get', 'dev/ping'), [
                 'query' => [
-                    'value' => $value ?? ((string) \time()),
+                    'value' => $params instanceof DevPingParams && is_string($value = $params->getValue())
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                    'value' => $params instanceof DevPingParams && \is_string($value = $params->getValue())
```

here and in all places (with "root namespaced" functions), please

File: src/Client.php [hunk 18]
```diff
@@ -157,12 +163,12 @@ public function user(bool $detailed = false): UserResponse
...
      */
-    public function userBalance(string $report_type_uid, bool $detailed = false): UserBalanceResponse
+    public function userBalance(UserBalanceParams $params): UserBalanceResponse
     {
         return UserBalanceResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', \sprintf('user/balance/%s', \urlencode($report_type_uid))), [
+            $this->doRequest(new Request('get', \sprintf('user/balance/%s', \urlencode($params->getReportTypeUid()))), [
                 'query' => [
-                    '_detailed' => $detailed
+                    '_detailed' => $params->isDetailed()
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                    '_detailed' => $params->isDetailed() === true
```

File: src/Client.php [hunk 19]
```diff
@@ -384,4 +416,24 @@ protected function dispatchEvent($event): void
...
     }
+
+    /**
+     * Common query part for requests `user/repors` and `user/report_types`.
+     *
+     * @param array<string, string|int>|null $additional_params
+     *
+     * @return array<string, string|int>
+     */
+    private function getListQueryProperties(?array $additional_params = null): array
```
Comment (tarampampam) [inline src/Client.php]: Remove this method. DRY here is not required

**Review by tarampampam — CHANGES_REQUESTED** [review]

File: src/Client.php (hunk 18 above)
Comment (gomzyakov) [reply src/Client.php]: Yep, I fixed.

At this point, we have a some different logic:
```php
    'query' => [
        '_content'  => $params->isIncludeContent() === false
            ? 'false'
            : 'true',
        '_detailed' => $params->isDetailed() === false
            ? 'false'
            : 'true',
    ]
```

File: src/Client.php (hunk 19 above)
Comment (gomzyakov) [reply src/Client.php]: Method `getListQueryProperties` was removed 721dac95e68fe6cafe39dc9aa6d5706f26358d44

File: src/Client.php (hunk 17 above)
Comment (gomzyakov) [reply src/Client.php]: Fixed in all places 18f0fa4b96cff68b4389afe9688d9beaa8a39878

File: src/Client.php [hunk 20]
```diff
@@ -173,87 +179,129 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
+            if (\is_string($filter_query = $params->getQuery())) {
+                $query['_query'] = $filter_query;
+            }
+
+            if (\is_int($per_page = $params->getPerPage())) {
+                $query['_size'] = \max(1, $per_page);
+            }
+
+            if (\is_int($offset = $params->getOffset())) {
+                $query['_offset'] = $offset;
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                $query['_offset'] = \max(0, $offset);
```

File: src/Client.php (hunk 18 above)
Comment (tarampampam) [reply src/Client.php]: It seems like not fixed now

File: src/Client.php [hunk 21]
```diff
@@ -173,87 +179,129 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                $query['_offset'] = \max(0, $offset);
```

File: src/Client.php [hunk 22]
```diff
@@ -173,87 +179,129 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
                 'query' => [
-                    '_content'  => $content
-                        ? 'true'
-                        : 'false',
-                    '_detailed' => $detailed
-                        ? 'true'
-                        : 'false',
+                    '_content'  => $params->isIncludeContent() === false
+                        ? 'false'
+                        : 'true',
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                    '_content'  => $params->isIncludeContent() === true
                        ? 'true'
                        : 'false',
```

File: src/Client.php [hunk 23]
```diff
@@ -173,87 +179,129 @@ public function userBalance(string $report_type_uid, bool $detailed = false): Us
...
+                    '_detailed' => $params->isDetailed() === false
+                        ? 'false'
+                        : 'true',
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                    '_detailed' => $params->isDetailed() === true
                        ? 'true'
                        : 'false',
```

File: src/Params/DevPingParams.php (hunk 6 above)
Comment (tarampampam) [inline src/Params/DevPingParams.php]: Question about parameter classes finalization still open - let's discuss about it tomorrow in DM

**Review by tarampampam — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 24]
```diff
@@ -14,6 +14,14 @@ The format is based on [Keep a Changelog][keepachangelog] and this project adher
 ### Changed
```
Comment (tarampampam) [inline CHANGELOG.md:22]: Where is `Added` section with description about all added classes?

File: src/Client.php (hunk 22 above)
Comment (gomzyakov) [reply src/Client.php]: We need the following behavior:

| $params->isIncludeContent() | '_content' |
| ---  | --- |
| true | true |
| false | false |
| null | true |

In the suggested code, the behavior is:

| $params->isIncludeContent() | '_content' |
| ---  | --- |
| true | true |
| false | false |
| **null** | **false** |

Property `_content` setted incorrect, when `isIncludeContent()` return `null`.

*P.S. But yes, the code looks strange*

File: src/Client.php (hunk 23 above)
Comment (gomzyakov) [reply src/Client.php]: Here's the same logic as in the [comment above](https://github.com/avtocod/b2b-api-php/pull/22#discussion_r509342841).

File: CHANGELOG.md (hunk 24 above)
Comment (gomzyakov) [reply CHANGELOG.md:22]: Sorry, I forgot. Added 29caa24043c1ef560e153aa938561ff08d7fd793

File: CHANGELOG.md [hunk 25]
```diff
@@ -8,12 +8,28 @@ The format is based on [Keep a Changelog][keepachangelog] and this project adher
 
 ### Added
 
+- Class `DevPingParams` as parameter to test connection
+- Class `DevTokenParams` as parameter to debug token generation
 - Class `ReportMakeParams` to build make-report parameters
```
Comment (tarampampam) [inline CHANGELOG.md]:
```suggestion
- Class `UserReportMakeParams` to build make-report parameters
```

File: src/Client.php [hunk 26]
```diff
@@ -157,12 +163,12 @@ public function user(bool $detailed = false): UserResponse
...
      */
-    public function userBalance(string $report_type_uid, bool $detailed = false): UserBalanceResponse
+    public function userBalance(UserBalanceParams $params): UserBalanceResponse
     {
         return UserBalanceResponse::fromHttpResponse(
-            $this->doRequest(new Request('get', \sprintf('user/balance/%s', \urlencode($report_type_uid))), [
+            $this->doRequest(new Request('get', \sprintf('user/balance/%s', \urlencode($params->getReportTypeUid()))), [
                 'query' => [
-                    '_detailed' => $detailed
+                    '_detailed' => $params->isDetailed() === true
```
Comment (tarampampam) [inline src/Client.php:171]:
```suggestion
                    '_detailed' => $params->isDetailed() === true || $params->isDetailed() === null // detailed by default
```

File: src/Client.php [hunk 27]
```diff
@@ -117,22 +127,18 @@ public function getVersion(bool $without_hash = true): string
...
     {
         return DevTokenResponse::fromHttpResponse(
             $this->doRequest(new Request('get', 'dev/token'), [
                 'query' => [
-                    'user'    => $username,
-                    'pass'    => $password,
-                    'is_hash' => $is_hash
+                    'user'    => $params->getUsername(),
+                    'pass'    => $params->getPassword(),
+                    'is_hash' => $params->isPasswordHashed()
```
Comment (tarampampam) [inline src/Client.php]:
```suggestion
                    'is_hash' => $params->isPasswordHashed() === true
```

Here and in all places, where method can return `nullable` value

**Review by tarampampam — CHANGES_REQUESTED** [review]

File: src/Client.php (hunk 27 above)
Comment (gomzyakov) [reply src/Client.php]: I went through the code. This is the only such place where a more stringent condition is required.

Fixed 166276ba317b164c336c5ff7ce3357a53a49fb74

**Review by tarampampam — APPROVED** [review]
